pmd-cs/src/test/resources/net/sourceforge/pmd/lang/cs/cpd/testdata/listOfNumbers.cs

```diff
@@ -0,0 +1,8 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+public class LongLists {
+    List<byte> l = new List<byte> {
```

Comment:
Isn't there a library method to initialize a list with copies of the same value? In Java you can write eg `new ArrayList<>(Collections.nCopies(120, 0))`.

Edit:

Found it: https://stackoverflow.com/a/17169142/6245827

So 2 questions:
* are there some cases where this AddRange method is inappropriate, and you'd really like to duplicate your literals?
* if so, then in those cases, isn't it more informative to CPD-OFF locally, and explain why you're not using AddRange with a comment, rather than use a global `--ignore-literal-sequences`?

Reply:
Thank you for your quick response and insight!

* You're right that this example isn't completely realistic. A more realistic case would be a large table with all kinds of numbers, with parts of them being zeroes. For low values of the minimum number of tokens, it's quite easy to reach the limit for a duplication then.

An example I can think of: transformation matrices for n-dimensional coordinates. A simple matrix may look as follows:
```
List<byte> scaleByTwo = new List<byte> {
2,0,0,0,0,0,0,0,0,0,
0,2,0,0,0,0,0,0,0,0,
0,0,2,0,0,0,0,0,0,0,
0,0,0,2,0,0,0,0,0,0,
0,0,0,0,2,0,0,0,0,0,
0,0,0,0,0,2,0,0,0,0,
0,0,0,0,0,0,2,0,0,0,
0,0,0,0,0,0,0,2,0,0,
0,0,0,0,0,0,0,0,2,0,
0,0,0,0,0,0,0,0,0,2,
};
```

The sequences of zeroes in this matrix already cause 21 duplicate tokens. While you could replace the sequences of zeroes by invocations of `AddRange`, I'd argue this hurts readability.
(Of course, you could generate this list as well. The point is that there may be examples where it becomes harder to do so or where generating or reading from a file causes unacceptable performance degredations.)

* On using CPD-OFF instead of recognizing these cases from CPD: this would also be a good solution, but it may cause frustration in domains where these lookup tables are common. We would like to avoid that developers start ignoring CPD output because they believe all duplication instances are simply lookup tables for which a real improvement is not possible, which could mean they leave in pieces of actual functionality (code) that are duplicated and will cause maintainability problems in the future.

Reply:
Ah ok I see. I think it's better style to generate such a matrix but I get that it may be frustrating to have irrelevant violations.

I wonder though, if this is the best way to fix it. Duplicating `0,` is indeed not really a problem, but duplicating the entire list initializer is. If we just remove all those tokens, there is no way we can find such a duplicate (and conversely, we may consider `List<> { }` and `List<> { 0,0,0, }` to be duplicates). But this is maybe a more general issue in CPD and not specific to this PR, so from my perspective your changes are ok to merge

Reply:
Interesting observation. I think you're correct that this can lead to false positives between empty and non-empty initializers. In my estimation, the overall impact is still positive, but this may be something to tackle better in the future.